src/metabase/models/emitter.clj

```diff
@@ -27,14 +41,15 @@
          {:primary-key (constantly :emitter_id)}))
 
 (defn- pre-insert
-  [emitter]
+  [{action-id :action_id, :as emitter}]
+  {:pre [(integer? action-id)]}
```

Comment:
Adding this assertion actually caught a lot of broken stuff in the tests (I will explain below)